File: benchmarks/pandas/bench_dataframe_count.py
"""Benchmark: DataFrame.count() on 100k-row DataFrame with some NAs."""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 5
ITERATIONS = 30

a = np.where(np.arange(ROWS) % 3 == 0, np.nan, np.arange(ROWS, dtype=float))
b = np.where(np.arange(ROWS) % 5 == 0, np.nan, np.arange(ROWS, dtype=float) * 2)
c = np.arange(ROWS, dtype=float) * 3
df = pd.DataFrame({"a": a, "b": b, "c": c})
for _ in range(WARMUP): df.count()

times = []
for _ in range(ITERATIONS):
    t0 = time.perf_counter()
    df.count()
    times.append(time.perf_counter() - t0)
total = sum(times) * 1000
print(json.dumps({ "function": "dataframe_count", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total }))

File: benchmarks/tsb/bench_dataframe_count.ts
/**
 * Benchmark: DataFrame.count() on 100k-row DataFrame with some NAs.
 */
import { DataFrame } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 5;
const ITERATIONS = 30;

const df = DataFrame.fromColumns({
  a: Array.from({ length: ROWS }, (_, i) => i % 3 === 0 ? null : i * 1.0),
  b: Array.from({ length: ROWS }, (_, i) => i % 5 === 0 ? null : i * 2.0),
  c: Array.from({ length: ROWS }, (_, i) => i * 3.0),
});

for (let i = 0; i < WARMUP; i++) df.count();

const times: number[] = [];
for (let i = 0; i < ITERATIONS; i++) {
  const t0 = performance.now();
  df.count();
  times.push(performance.now() - t0);
}
const total = times.reduce((a, b) => a + b, 0);
console.log(JSON.stringify({ function: "dataframe_count", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));
